tests/test_slugs.py

```diff
@@ -40,7 +39,27 @@ def test_old_slugs(self):
         foo.slugs.update(is_active=True)
         assert foo.old_slugs == ["foold"]
 
+    def test_prefetch_related(self, django_assert_max_num_queries):
+        for i in range(10):
+            # 10 Foo instances with two slugs each
+            foo = Foo.objects.create(name=f"Foo {i}")
+            foo.ensure_slug()
+            foo.name = f"Bar {i}"
+            foo.save()
+            foo.ensure_slug()
+
+        with django_assert_max_num_queries(2):
+            # One query for all Foos and another for all FooSlugs
+            qs = Foo.objects.prefetch_related("slugs")
+            results = [foo.pk for foo in qs]
+            slugs = [foo.slug for foo in qs]
+            old_slugs = [foo.old_slugs for foo in qs]
+
+        assert len(results) == 10
+        assert len(slugs) == 10
+        assert len(old_slugs) == 10
+
 
 def test_slug_str():
-    slug = AbstractSlug(slug="nop")
```

Comment:
Django complained here: `TypeError: Abstract models cannot be instantiated`. Switched to the concrete model instead.